creator-node/test/CNodeHealthManager.test.js

```diff
@@ -0,0 +1,478 @@
+  })
+
+  it('returns true when health check passes with performSimpleCheck=true', async function () {
+    // Stub functions that isNodeHealthy() will call
+    const node = 'http://some_content_node.co'
+    const verboseHealthCheckResp = { healthy: true }
+    const queryVerboseHealthCheckStub = sandbox
+      .stub(CNodeHealthManager, 'queryVerboseHealthCheck')
+      .resolves(verboseHealthCheckResp)
+    const determinePeerHealthStub = sandbox.stub(
+      CNodeHealthManager,
+      'determinePeerHealth'
+    )
+    const logErrorStub = sandbox.stub(CNodeHealthManager, 'logError')
+
+    // Verify that only the simple check was performed and returned healthy
+    const isHealthy = await CNodeHealthManager.isNodeHealthy(node, true)
+    expect(isHealthy).to.be.true
+    expect(queryVerboseHealthCheckStub).to.have.been.calledOnceWithExactly(node)
+    expect(determinePeerHealthStub).to.not.have.been.called
+    expect(logErrorStub).to.not.have.been.called
+  })
+
+  for (const performSimpleCheck of [true, false]) {
+    it(`returns false when health check fails with performSimpleCheck=${performSimpleCheck}`, async function () {
+      // Stub functions that isNodeHealthy() will call
+      const node = 'http://some_content_node.co'
+      const error = new Error('test error')
+      const queryVerboseHealthCheckStub = sandbox
+        .stub(CNodeHealthManager, 'queryVerboseHealthCheck')
+        .rejects(error)
+      const determinePeerHealthStub = sandbox.stub(
+        CNodeHealthManager,
+        'determinePeerHealth'
+      )
+      const logErrorStub = sandbox.stub(CNodeHealthManager, 'logError')
+
+      // Verify that only the simple check was performed and returned healthy
+      const isHealthy = await CNodeHealthManager.isNodeHealthy(
+        node,
+        performSimpleCheck
+      )
+      expect(isHealthy).to.be.false
+      expect(queryVerboseHealthCheckStub).to.have.been.calledOnceWithExactly(
+        node
+      )
+      expect(determinePeerHealthStub).to.not.have.been.called
+      expect(logErrorStub).to.have.been.called.calledOnceWithExactly(
+        `isNodeHealthy() peer=${node} is unhealthy: ${error.toString()}`
+      )
+    })
+  }
+
+  it('returns false when determinePeerHealth throws with performSimpleCheck=false', async function () {
+    // Stub functions that isNodeHealthy() will call
+    const node = 'http://some_content_node.co'
+    const determinePeerHealthError = new Error('test determinePeerHealthError')
+    const verboseHealthCheckResp = { healthy: false }
+    const queryVerboseHealthCheckStub = sandbox
+      .stub(CNodeHealthManager, 'queryVerboseHealthCheck')
+      .resolves(verboseHealthCheckResp)
+    const determinePeerHealthStub = sandbox
+      .stub(CNodeHealthManager, 'determinePeerHealth')
+      .throws(determinePeerHealthError)
+    const logErrorStub = sandbox.stub(CNodeHealthManager, 'logError')
+
+    // Verify that only the simple check was performed and returned healthy
```

Comment:
by `returned healthy`, did you mean returned 200 status code but the node stats themselves would cause determinePeerHealth to fail?

Reply:
not sure what I meant but that comment doesn't seem accurate. I'll fix it